app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -299,96 +299,104 @@ def vms
-        flavor.try(:disk).to_i.gigabytes + flavor.try(:swap).to_i.megabytes + flavor.try(:ephemeral).to_i.gigabytes
-      )
-      hardware.guest_os = persister.hardwares.lazy_find(miq_template_lazy, :key => :guest_os)
-
-      operating_system = persister.operating_systems.find_or_build(server)
-      operating_system.vm_or_template = server
-      operating_system.product_name = persister.operating_systems.lazy_find(miq_template_lazy, :key => :product_name)
-      operating_system.distribution = persister.operating_systems.lazy_find(miq_template_lazy, :key => :distribution)
-      operating_system.version = persister.operating_systems.lazy_find(miq_template_lazy, :key => :version)
-
-      attachment_names = {'vda' => 'Root disk'}
-      disk_location = "vda"
-      if (root_size = flavor.try(:disk).to_i.gigabytes).zero?
-        root_size = 1.gigabytes
-      end
-      make_instance_disk(hardware, root_size, disk_location.dup, attachment_names[disk_location])
-      ephemeral_size = flavor.try(:ephemeral).to_i.gigabytes
-      unless ephemeral_size.zero?
-        disk_location = "vdb"
-        attachment_names[disk_location] = "Ephemeral disk"
-        make_instance_disk(hardware, ephemeral_size, disk_location, attachment_names[disk_location])
-      end
-      swap_size = flavor.try(:swap).to_i.megabytes
-      unless swap_size.zero?
-        disk_location = disk_location.succ
-        attachment_names[disk_location] = "Swap disk"
-        make_instance_disk(hardware, swap_size, disk_location, attachment_names[disk_location])
-      end
+      vm_details(s, parent_host, parent_cluster)
+    end
+  end
 
-      # Make disks in the inventory for each of this server's volume attachments.
-      # Start by checking the raw attributes from fog to see whether whether this
-      # server has any attachments. If it does, then use the fog object's
-      # volume_attachments method to inflate them and get the device names.
-      # Checking the attribute first avoids making an expensive api call for
-      # every server when they may not all have attachments.
-      # Don't worry about filling in the volume, since the volume service refresh
-      # will take care of that.
-      if s.attributes.fetch("os-extended-volumes:volumes_attached", []).length > 0
-        s.volume_attachments.each do |attachment|
-          # Skip Volume mounts without mount point
-          next if attachment['device'].blank?
-
-          dev = File.basename(attachment['device'])
-          persister.disks.find_or_build_by(
-            :hardware    => hardware,
-            # reuse the device names from above in the event that this is an
-            # instance that was booted from a volume
-            :device_name => attachment_names.fetch(dev, dev)
-          ).assign_attributes(
-            :location        => dev,
-            :device_type     => "disk",
-            :controller_type => "openstack"
-          )
-        end
+  def get_flavor(vm)
```

Comment:
Where is this method used?

Reply:
Looks like only in `ManageIQ::Providers::IbmPowerVc::Inventory::Parser::CloudManager` so we should move it to that class:
https://github.com/ManageIQ/manageiq-providers-ibm_power_vc/pull/28/files#diff-2dbea85c89843b8aa05cba1d5a6caa08e2b2389d21580c2176026180d415727eR7

Reply:
Actually looks like it could be used down [[here]](https://github.com/ManageIQ/manageiq-providers-openstack/pull/753/files#diff-46c42db89009bb02993acce873dd594e15baf153ebda72079095b9946a480802R331)